### PR TITLE
feat: add buy/sale websocket events

### PR DESCRIPTION
**Pull request by d3or**

### CODE REVIEW

File: packages/indexer/src/jobs/websocket-events/buy-websocket-events-trigger-queue.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+          txHash: r.tx_hash,
+          logIndex: r.log_index,
+          batchIndex: r.batch_index,
+          isDeleted: Boolean(r.is_deleted),
+          createdAt: new Date(r.created_at).toISOString(),
+          updatedAt: new Date(r.updated_ts * 1000).toISOString(),
+        });
+
+        let eventType = "";
+        if (data.trigger === "insert") eventType = "buy.created";
```
Comment (nofir) [inline packages/indexer/src/jobs/websocket-events/buy-websocket-events-trigger-queue.ts:124]: What is buy? Why dont just have a sale websocket? we have the side and the taker and the maker.

File: packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-queue.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+          orderKind: r.order_kind,
+          maker: r.maker,
+          taker: r.taker,
+          amount: r.amount,
+          fillSourceId: r.fill_source_id,
+          block: r.block,
+          txHash: r.tx_hash,
+          logIndex: r.log_index,
+          batchIndex: r.batch_index,
+          isDeleted: Boolean(r.is_deleted),
```
Comment (nofir) [inline packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-queue.ts:118]: Should we created a sale.deleted / sale.removed based on is_deleted? wdyt?

Comment (d3or) [reply packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-queue.ts:118]: in what instances are is_deleted set to true?

Comment (nofir) [reply packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-queue.ts:118]: after we mark it as deleted (usually when we detect an issue or when we revert sales due to orphan blocks)

**Review by nofir — APPROVED** [review]
approved but lets see if there is feedback regarding the payload

File: packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-queue.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
```
Comment (nofir) [inline packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-queue.ts:118]: you should remove this from the payload.

**Review by nofir — CHANGES_REQUESTED** [review]

**Pull request merged**